### ISSUE TITLE
`better-regex` should mark wrapping groups for removal

### ISSUE THREAD

**Issue by ST-DDT** [sindresorhus/eslint-plugin-unicorn#2202]
The `better-regex` rule should mark wrapping groups for removal.

**This should fail**

````ts
/(test)/
/^(test)$/
/(^test)$/
/(^test$)/
/^(test$)/
````
`test` can be anything as long as it isn't an immediate union child (``|``) of the wrapping group.

**This does and should pass**

````ts
/test/
/^test$/
/(te)(st)/
/^(te)(st)$/
/^(te|st)$/
/(^te|st)$/
/^(te|st$)/
````

**Reasoning**

The wrapping group (`result[1]`) is the same as `result[0]`.

**Limitations**

This should probably not be auto fixable to avoid changing the indexes of the result array.

**ToDos**

Check how the regex flags such as `/gml` affect the output of the regex.

**Comment by fisker** [comment]
Have you checked [`eslint-plugin-regexp`](https://github.com/ota-meshi/eslint-plugin-regexp)? Maybe there already a rule for this.

**Comment by ST-DDT** [comment]
I havent checked there because the recommended configs of both plugins conflict.

- https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1852

**Comment by fisker** [comment]
`better-regex` can be useful for users not using other  regexp related rules. I'd suggest disable it yourself if you find conflicts.

**Comment by fregante** [comment]
I think [`better-regex`](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/c035216f1ed6e3e82b985e057d06bd0dabab0513/rules/better-regex.js) is just a wrapper around the packages:

- regexp-tree
- clean-regexp

So either propose it to those dependencies or use the dedicated regex plugin suggested
